cpp/src/arrow/extension/fixed_shape_tensor.cc

```diff
@@ -340,6 +391,12 @@ Result<std::shared_ptr<DataType>> FixedShapeTensorType::Make(
     return Status::Invalid("dim_names size must match shape size. Expected: ",
                            shape.size(), " Got: ", dim_names.size());
   }
+  for (auto i : permutation) {
+    if (i < 0 || i >= static_cast<int64_t>(shape.size())) {
+      return Status::Invalid("permutation indices must be in [0, shape.size()). Got: ",
+                             i);
+    }
```

Comment:
Is that all? Am I allowed to pass `[1,2,2,1]` as permutation?

Reply:
Changed the check and added a test: https://github.com/apache/arrow/pull/37533/commits/37b398e5d266c0543b366513e4a7b42641ca6f94